internal/server/monitoring.go

```diff
@@ -0,0 +1,55 @@
+package server
+
+import (
+	"context"
+	"time"
+
+	"github.com/clems4ever/go-graphkb/internal/knowledge"
+	"github.com/clems4ever/go-graphkb/internal/metrics"
+	"github.com/clems4ever/go-graphkb/internal/sources"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+func getMonitoringIntervalSeconds() time.Duration {
+	var interval time.Duration
+	intervalSeconds := viper.GetInt("monitoring_interval_seconds")
+	if intervalSeconds == 0 {
+		interval = 60 * time.Second
+	} else {
+		interval = time.Duration(intervalSeconds) * time.Second
+	}
+	return interval
+}
+
+// startGraphSizeMonitoring start a background process regularly checking the size of the graph to expose it as a prometheus metric
+func startGraphSizeMonitoring(interval time.Duration, database knowledge.GraphDB, sourcesRegistry sources.Registry) {
+	logrus.Infof("Monitoring of the graph size will happen every %ds", int(interval/time.Second))
+	go func() {
+		ctx := context.Background()
+		for {
+			logrus.Debug("Start monitoring the graph size...")
+			sources, err := sourcesRegistry.ListSources(ctx)
+			if err != nil {
+				logrus.Errorf("Unable to list sources for monitoring: %v", err)
+				continue
```

Comment:
These `continue`s are dangerous the way they are setup, metrics won't get updated on error

Reply:
At least they will keep their old value which is safe to assume considering how alerts work I guess.
What's your suggestion otherwise?

Reply:
Clear, or set to 0 the metrics. If an importer add truckloads of nodes and suddenly these queries fail, these metrics will mislead us with the previous values.

Reply:
last remaining question. What do you think?